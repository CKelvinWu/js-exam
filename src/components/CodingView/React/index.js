import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import 'brace';
import 'brace/mode/jsx';
import 'brace/mode/javascript';
import 'brace/theme/textmate';
import 'brace/theme/monokai';

import Grid from 'components/Grid';
import GridItem from 'components/Grid/GridItem';
import ConsoleWidget from 'components/Widgets/ConsoleWidget';
import CodeWidget from 'components/Widgets/CodeWidget';
import ResultWidget from 'components/Widgets/ResultWidget';
import AnswerWidget from 'components/Widgets/AnswerWidget';

import debouncedRunCode from 'utils/runCode';
import { REACT as GRID_LABEL_REACT } from 'utils/gridLabel';

import styles from './ReactPage.module.scss';

class ReactPage extends Component {
  static propTypes = {
    isExaming: PropTypes.bool,
    code: PropTypes.string,
    compiledCode: PropTypes.string,
    consoleMsg: PropTypes.array,
    handleCodeChange: PropTypes.func,
    wrappedConsole: PropTypes.object,
    resetConsole: PropTypes.func,
  };

  controlHeight = 70;

  componentDidMount() {
    const { compiledCode, wrappedConsole, resetConsole } = this.props;
    resetConsole();
    debouncedRunCode({ code: compiledCode, wrappedConsole });
  }

  shouldComponentUpdate(nextProps) {
    const { compiledCode: previousCompiledCode } = this.props;
    const { compiledCode, wrappedConsole, resetConsole } = nextProps;
    if (previousCompiledCode !== compiledCode) {
      resetConsole();
      // WARNING: This is not debounced
      debouncedRunCode({ code: compiledCode, wrappedConsole });
    }
    return true;
  }

  render() {
    const { isExaming, code, handleCodeChange, consoleMsg } = this.props;
    const layout = [
      {
        key: 'code',
        x: 0,
        y: 0,
        width: window.innerWidth / 2 + 50,
        height: window.innerHeight / 2 + 100,
        minWidth: 100,
        minHeight: 100,
        maxWidth: 700,
        maxHeight: 500,
      },
      {
        key: 'answer',
        x: 0,
        y: 1,
        width: window.innerWidth / 2 + 50,
        height: window.innerHeight / 2 - 150,
        minWidth: 100,
        maxWidth: 700,
      },
      {
        key: 'result',
        x: 1,
        y: 0,
        width: window.innerWidth / 2 - 50,
        height: (window.innerHeight - this.controlHeight) / 2,
        minWidth: 100,
        minHeight: 100,
        maxWidth: 700,
        maxHeight: 500,
      },
      {
        key: 'console',
        x: 1,
        y: 1,
        width: window.innerWidth / 2 - 50,
        height: (window.innerHeight - this.controlHeight) / 2 - 50,
        minWidth: 100,
        minHeight: 100,
        maxWidth: 700,
        maxHeight: 500,
      },
    ];
    return (
      <div className={styles.app}>
        <Grid layout={layout} totalWidth="100%" totalHeight="100%" autoResize>
          <GridItem key="code" label={GRID_LABEL_REACT.code}>
            <CodeWidget
              handleCodeChange={handleCodeChange}
              data={code}
              readOnly={!isExaming}
              mode="jsx"
              theme="monokai"
            />
          </GridItem>
          <GridItem key="answer" label={GRID_LABEL_REACT.answer}>
            {isExaming ? <AnswerWidget /> : null}
          </GridItem>
          <GridItem key="result" label={GRID_LABEL_REACT.result}>
            {isExaming ? <ResultWidget /> : null}
          </GridItem>
          <GridItem key="console" label={GRID_LABEL_REACT.console}>
            {isExaming ? <ConsoleWidget data={consoleMsg} /> : null}
          </GridItem>
        </Grid>
      </div>
    );
  }
}

export default withRouter(ReactPage);
