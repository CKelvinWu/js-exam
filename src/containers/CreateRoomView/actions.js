/*
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */
import { API, graphqlOperation } from 'aws-amplify';
import * as mutations from 'graphql/mutations';
import { action } from 'utils/actionsHelper';
import { message } from 'antd';
import { CREATE_ROOM } from './constants';

export const createRoomActions = {
  request: data => action(CREATE_ROOM.REQUEST, { data }),
  success: data => action(CREATE_ROOM.SUCCESS, { data }),
  failure: error => action(CREATE_ROOM.FAILURE, { error }),
};

export function createRoom(data) {
  return async dispatch => {
    dispatch(createRoomActions.request(data));
    try {
      const { data: testData } = await API.graphql(graphqlOperation(mutations.createTest, data));
      const roomNum = Math.floor(Math.random() * 98) + 1;
      const roomChar = String.fromCharCode(Math.floor(Math.random() * 26) + 65);
      const { data: resData } = await API.graphql(
        graphqlOperation(mutations.createRoom, {
          input: {
            roomTestId: testData.createTest.id,
            description: roomChar + roomNum,
            createTime: new Date(),
            ...data.input,
          },
        }),
      );
      dispatch(createRoomActions.success(resData.createRoom));
    } catch (error) {
      dispatch(createRoomActions.failure(error));
      message.error('Create room failed');
    }
  };
}
