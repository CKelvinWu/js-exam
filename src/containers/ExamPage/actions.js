import { API, graphqlOperation } from 'aws-amplify';
import * as mutations from 'graphql/mutations';

export function updateRecordData(data) {
  return async dispatch => {
    try {
      const params = {
        input: {
          id: data.id,
          syncCode: data.newCode,
          timeEnd: new Date(),
        },
      };
      await API.graphql(graphqlOperation(mutations.updateRecord, params));
      dispatch(createHistory({ historyData: data.newCode }));
    } catch (error) {
      console.log(error);
    }
  };
}

function createHistory(data) {
  return async (dispatch, getState) => {
    const { id: recordId } = getState().record;
    try {
      const createTime = new Date();
      await API.graphql(
        graphqlOperation(mutations.createHistory, {
          input: {
            time: createTime,
            code: data.historyData,
            historyRecordId: recordId,
          },
        }),
      );
    } catch (error) {
      console.log(error);
    }
  };
}
