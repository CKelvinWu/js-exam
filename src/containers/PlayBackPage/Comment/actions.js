import { API, graphqlOperation } from 'aws-amplify';
import { queryRecordWithHistory } from './queries';

export function fetchRecordWithHistory(id) {
  return async (dispatch, getState) => {
    try {
      const {
        history: { nextToken },
      } = getState().record;

      const query = {
        id,
        limit: 1000,
        nextToken,
      };
      const { data } = await API.graphql(
        graphqlOperation(queryRecordWithHistory, query),
      );

      const histories = data.getRecord.history.items;
      const result = {
        history: sortByTime(histories),
        ...data.getRecord,
      };
      console.log(result)
      dispatch(setCurrentRecordWithHistory(result));
    } catch (e) {
      console.log(e);
    }
  };
}

function setCurrentRecordWithHistory(result) {
  return {
    type: 'SET_CURRENT_RECORD_WITH_HISTORY',
    payload: result,
  };
}

function sortByTime(data) {
  return data.sort(
    (a, b) => new Date(a.time).getTime() - new Date(b.time).getTime(),
  );
}
