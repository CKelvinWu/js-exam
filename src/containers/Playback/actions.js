import { API, graphqlOperation } from 'aws-amplify';
import { queryRecordWithHistory } from './queries';

export function fetchRecordWithHistory(id) {
  return async (dispatch, getState) => {
    try {
      const {
        history: { nextToken },
      } = getState().record;

      const query = {
        id,
        limit: 1000,
        nextToken,
      };
      const { data } = await API.graphql(
        graphqlOperation(queryRecordWithHistory, query),
      );
      console.log('#getRecordWithHistory', data);

      dispatch(setCurrentRecordWithHistory(data.getRecord));
    } catch (e) {
      console.log(e);
    }
  };
}

function setCurrentRecordWithHistory(result) {
  return {
    type: 'SET_CURRENT_RECORD_WITH_HISTORY',
    payload: result,
  };
}
