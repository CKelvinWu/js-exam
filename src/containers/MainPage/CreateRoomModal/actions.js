/*
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */
import { API, graphqlOperation } from 'aws-amplify';
import * as mutations from 'graphql/mutations';
import { action } from 'utils/actionsHelper';
import { message } from 'antd';
import { setHostings } from 'redux/login/actions';
import { CREATE_ROOM } from './constants';

export const createRoomActions = {
  reset: () => action(CREATE_ROOM.RESET),
  request: data => action(CREATE_ROOM.REQUEST, { data }),
  success: data => action(CREATE_ROOM.SUCCESS, { data }),
  failure: error => action(CREATE_ROOM.FAILURE, { error }),
};

const createRoomQl = `mutation CreateRoom($input: CreateRoomInput!) {
  createRoom(input: $input) {
    id
    test {
      id
      subjectId
      description
      timeBegin
      timeEnd
      status
      tags
      host{
        id
        name
      }
    }
    subjectId
    description
    host {
      id
      name
    }
    createTime
    password
    users {
      items {
        id
        name
      }
      nextToken
    }
    currentRecord {
      id
      subjectId
      syncCode
      timeBegin
      timeEnd
      status
    }
  }
}
`;

export function createRoom(data) {
  return async dispatch => {
    dispatch(createRoomActions.request(data));
    try {
      const roomNum = Math.floor(Math.random() * 98) + 1;
      const roomChar = String.fromCharCode(Math.floor(Math.random() * 26) + 65);
      const createTime = new Date();
      // TODO: change graphql schema such that a jeUser can host mutiple room and test
      const { data: jeUserData } = await API.graphql(
        graphqlOperation(mutations.createJeUser, {
          input: {
            name: localStorage.username,
          },
        }),
      );
      const { data: testData } = await API.graphql(
        graphqlOperation(mutations.createTest, {
          input: {
            timeBegin: createTime,
            status: 'open',
            testHostId: jeUserData.createJEUser.id,
            // temporarily store host name in tags, since a jeUser cannot host mutiple test in current graphql schema
            tags: localStorage.username,
            ...data,
          },
        }),
      );
      const { data: roomData } = await API.graphql(
        graphqlOperation(createRoomQl, {
          input: {
            roomTestId: testData.createTest.id,
            description: roomChar + roomNum,
            createTime,
            ...data,
          },
        }),
      );
      // TODO: change graphql schema to connect jeUser and test and room. So that don't have to update manually.
      const { data: jeUserUpdate } = await API.graphql(
        graphqlOperation(mutations.updateJeUser, {
          input: {
            id: jeUserData.createJEUser.id,
            jEUserRoomId: roomData.createRoom.id,
            jEUserTestId: testData.createTest.id,
            jEUserHostTestId: testData.createTest.id,
          },
        }),
      );
      const { data: testUpdate } = await API.graphql(
        graphqlOperation(mutations.updateTest, {
          input: {
            id: testData.createTest.id,
            testRoomId: roomData.createRoom.id,
          },
        }),
      );

      dispatch(createRoomActions.success(roomData.createRoom));

      /* add a hosting room into 'hostings' array in local storage */
      dispatch(setHostings(roomData.createRoom.id));
    } catch (error) {
      dispatch(createRoomActions.failure(error));
      message.error('Create room failed');
      console.log(error);
    }
  };
}
