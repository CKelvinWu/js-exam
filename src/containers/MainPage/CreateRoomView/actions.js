/*
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */
import { API, graphqlOperation } from 'aws-amplify';
import * as mutations from 'graphql/mutations';
import { action } from 'utils/actionsHelper';
import { message } from 'antd';
import { setHostings } from 'redux/login/actions';
import { CREATE_ROOM } from './constants';

export const createRoomActions = {
  request: data => action(CREATE_ROOM.REQUEST, { data }),
  success: data => action(CREATE_ROOM.SUCCESS, { data }),
  failure: error => action(CREATE_ROOM.FAILURE, { error }),
};

export function createRoom(data) {
  return async dispatch => {
    dispatch(createRoomActions.request(data));
    try {
      const roomNum = Math.floor(Math.random() * 98) + 1;
      const roomChar = String.fromCharCode(Math.floor(Math.random() * 26) + 65);
      const createTime = new Date();
      const { data: testData } = await API.graphql(
        graphqlOperation(mutations.createTest, {
          input: {
            timeBegin: createTime,
            status: 'open',
            ...data,
          },
        }),
      );
      const { data: roomData } = await API.graphql(
        graphqlOperation(mutations.createRoom, {
          input: {
            roomTestId: testData.createTest.id,
            description: roomChar + roomNum,
            createTime,
            ...data,
          },
        }),
      );
      dispatch(createRoomActions.success(roomData.createRoom));

      /* add a hosting room into 'hostings' array in local storage */
      dispatch(setHostings(roomData.createRoom.id));
    } catch (error) {
      dispatch(createRoomActions.failure(error));
      message.error('Create room failed');
      console.log(error);
    }
  };
}
