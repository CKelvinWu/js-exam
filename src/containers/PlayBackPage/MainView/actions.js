import { API, graphqlOperation } from 'aws-amplify';
import { changeCode } from 'redux/code/actions';
import { queryRecordWithHistory } from './queries';
import {
  SET_CURRENT_RECORD_WITH_HISTORY,
  SET_SNAP_COMMENTS,
  SET_CATEGORY_INDEX,
  SET_RECORD_INDEX,
  SET_HISTORY_INDEX,
} from './constants';

export function fetchRecordWithHistory(id, index) {
  return async (dispatch, getState) => {
    try {
      const {
        history: { nextToken },
      } = getState().record;

      const query = {
        id,
        limit: 1000,
        nextToken,
      };
      const { data } = await API.graphql(
        graphqlOperation(queryRecordWithHistory, query),
      );

      const histories = data.getRecord.history.items;
      const result = {
        history: sortByTime(histories),
        ...data.getRecord,
      };
      dispatch(setHistoryIndex(0));
      dispatch(setCurrentRecordWithHistory(result));
      dispatch(setCategoryIndex(result.ques.type === 'javascript' ? 0 : 1));
      dispatch(setRecordIndex(index));
      if (result.history.items.length > 0) {
        dispatch(changeCode({ rawCode: result.history.items[0].code }));
        dispatch(setSnapComments(getSnapComments(result.history.items)));
      } else {
        dispatch(changeCode({ rawCode: result.ques.content }));
      }
      console.log(result);
    } catch (e) {
      console.log(e);
    }
  };
}

function setCurrentRecordWithHistory(result) {
  return {
    type: SET_CURRENT_RECORD_WITH_HISTORY,
    payload: result,
  };
}

function setSnapComments(snapComments) {
  return {
    type: SET_SNAP_COMMENTS,
    snapComments,
  };
}
export function setCategoryIndex(index) {
  return {
    type: SET_CATEGORY_INDEX,
    index,
  };
}
export function setRecordIndex(index) {
  return {
    type: SET_RECORD_INDEX,
    index,
  };
}

export function setHistoryIndex(index) {
  return {
    type: SET_HISTORY_INDEX,
    index,
  };
}

function sortByTime(data) {
  return data.sort(
    (a, b) => new Date(a.time).getTime() - new Date(b.time).getTime(),
  );
}

function getSnapComments(histories) {
  const snapComments = [];
  histories.forEach((history, i) => {
    if (history.snapComments.items.length > 0) {
      snapComments.push({
        historyIndex: i,
        data: history.snapComments.items,
      });
    }
  });
  return snapComments;
}
