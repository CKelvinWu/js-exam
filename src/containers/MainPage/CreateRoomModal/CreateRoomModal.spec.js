import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { message } from 'antd';
import configureStore from 'redux/configureStore';
import * as actions from 'redux/createRoomModal/actions';
import ConnectedCreateRoomModal, { CreateRoomModal } from './index';

describe('<CreateRoomModal />', () => {
  let wrapper;
  let subject;
  let instance;
  let dispatch;
  const username = 'user name';
  const createdRoomData = {
    id: 'f509ee49-c4d6-4d3c-b357-2d18454b01f2',
    subjectId: 'test room 1',
    description: 'Z77',
    createTime: '2020-01-31T07:07:03.295Z',
    createTimeByDate: new Date(
      'Fri Jan 31 2020 15:07:03 GMT+0800 (Taipei Standard Time)',
    ),
  };
  const propsFromParent = {
    visible: true,
    onClose: jest.fn(),
    history: { push: jest.fn() },
  };

  describe('before create room success', () => {
    beforeAll(() => {
      window.location.href = '/#/admin';
    });
    beforeEach(() => {
      jest.resetAllMocks();
      const store = configureStore({});
      store.dispatch = jest.fn(action => action);
      const props = { ...propsFromParent };
      wrapper = mount(
        <Provider store={store}>
          <BrowserRouter>
            <ConnectedCreateRoomModal {...props} />
          </BrowserRouter>
        </Provider>,
      );
      subject = wrapper.find('CreateRoomModal');
      dispatch = wrapper.prop('store').dispatch;
    });

    it('should render', () => {
      expect(wrapper.length).toBeTruthy();
    });

    it('should render as snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
      expect(subject.find('input').prop('placeholder')).toBe(
        'Please enter Candidate name',
      );
    });

    it('should handle user name input', () => {
      expect(subject.state()).toEqual({ name: '', disabled: true });
      const nameInput = subject.find('input');
      nameInput.simulate('change', { target: { value: username } });
      expect(subject.state()).toEqual({ name: username, disabled: false });
      nameInput.simulate('change', { target: { value: '' } });
      expect(subject.state()).toEqual({ name: '', disabled: true });
    });

    it('should handle ok', () => {
      jest.spyOn(actions, 'createRoom').mockImplementation(data => ({
        action: 'createRoom',
        ...data,
      }));
      subject.setState({ name: username, disabled: false });
      const CreateButton = subject.find('button').last();
      CreateButton.simulate('click');
      expect(dispatch).toBeCalledWith({
        action: 'createRoom',
        subjectId: username,
      });
    });

    it('should handle close', () => {
      const closeButton = subject.find('button').first();
      closeButton.simulate('click');
      expect(subject.prop('onClose')).toBeCalled();
      const cancelButton = subject.find('button').at(1);
      cancelButton.simulate('click');
      expect(subject.prop('onClose')).toBeCalledTimes(2);
    });
  });

  describe('after create room success', () => {
    beforeEach(() => {
      jest.resetAllMocks();
      dispatch = jest.fn(action => action);
      const stateProps = {
        isCreatingRoom: false,
        createRoomSuc: true,
        createdRoomData,
      };
      const dispatchProps = {
        onReset: jest.fn(),
        onCreateRoom: jest.fn(),
        onSetRoomHost: jest.fn(),
      };
      const props = {
        ...propsFromParent,
        ...stateProps,
        ...dispatchProps,
      };
      subject = mount(<CreateRoomModal {...props} />);
      subject.setState({ disabled: false, name: username });
      instance = subject.instance();
    });

    it('should render', () => {
      expect(subject.length).toBeTruthy();
    });

    it('should render as snapshot', () => {
      expect(subject.html()).toMatchSnapshot();
      expect(subject.find('p').text()).toBe('Candidate: test room 1');
    });

    it('should handle copy link', () => {
      document.execCommand = jest.fn();
      jest.spyOn(instance.examLinkInput.current, 'select');
      jest.spyOn(message, 'success');
      const copyLink = subject.find('input');
      expect(copyLink.prop('value')).toBe(
        `http://localhost/#/exam/${createdRoomData.id}`,
      );
      const copyLinkButton = subject.find('button.copyButton');
      copyLinkButton.simulate('click');
      expect(instance.examLinkInput.current.select).toBeCalled();
      expect(document.execCommand).toBeCalledWith('copy');
      expect(message.success).toBeCalledWith('Link has been copied.');
    });

    it('should handle enter room', () => {
      const handleOnAfterCloseSpy = jest
        .spyOn(instance, 'handleOnAfterClose')
        .mockImplementation(jest.fn());
      const { onSetRoomHost, history } = subject.props();
      const enterRoomButton = subject.find('button').last();
      enterRoomButton.simulate('click');
      expect(instance.handleOnAfterClose).toBeCalled();
      expect(onSetRoomHost).toBeCalledWith(true);
      expect(history.push).toBeCalledWith(
        `/admin/dispatch/${createdRoomData.id}`,
      );
      handleOnAfterCloseSpy.mockRestore();
    });

    it('should handle close', () => {
      const closeButton = subject.find('button').first();
      closeButton.simulate('click');
      expect(subject.prop('onClose')).toBeCalled();
      const cancelButton = subject.find('button').at(2);
      cancelButton.simulate('click');
      expect(subject.prop('onClose')).toBeCalledTimes(2);
    });

    it('should handle after close', () => {
      const onReset = subject.prop('onReset');
      instance.handleOnAfterClose();
      expect(subject.state()).toEqual({
        name: '',
        disabled: true,
      });
      expect(onReset).toBeCalled();
    });
  });
});
