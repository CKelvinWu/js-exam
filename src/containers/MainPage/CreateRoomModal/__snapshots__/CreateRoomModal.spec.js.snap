// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRoomModal /> after create room success should render as snapshot 1`] = `"<div class=\\"ant-modal-root\\"><div class=\\"ant-modal-mask fade-appear\\"></div><div tabindex=\\"-1\\" class=\\"ant-modal-wrap \\" role=\\"dialog\\" aria-labelledby=\\"rcDialogTitle6\\"><div role=\\"document\\" style=\\"width: 520px;\\" class=\\"ant-modal zoom-appear\\"><div tabindex=\\"0\\" style=\\"width: 0px; height: 0px; overflow: hidden;\\" aria-hidden=\\"true\\"></div><div class=\\"ant-modal-content\\"><button type=\\"button\\" aria-label=\\"Close\\" class=\\"ant-modal-close\\"><span class=\\"ant-modal-close-x\\"><i aria-label=\\"icon: close\\" class=\\"anticon anticon-close ant-modal-close-icon\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"close\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z\\"></path></svg></i></span></button><div class=\\"ant-modal-header\\"><div class=\\"ant-modal-title\\" id=\\"rcDialogTitle6\\">Welcome to Room - Z77</div></div><div class=\\"ant-modal-body\\"><p>Candidate: test room 1</p><div class=\\"content\\"><input readonly=\\"\\" type=\\"text\\" class=\\"ant-input ant-input-lg\\" value=\\"http://localhost//f509ee49-c4d6-4d3c-b357-2d18454b01f2\\"><button type=\\"button\\" class=\\"ant-btn copyButton\\"><i aria-label=\\"icon: copy\\" class=\\"anticon anticon-copy\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"copy\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z\\"></path></svg></i></button></div></div><div class=\\"ant-modal-footer\\"><div><button type=\\"button\\" class=\\"ant-btn\\"><span>Cancel</span></button><button type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Enter Room</span></button></div></div></div><div tabindex=\\"0\\" style=\\"width: 0px; height: 0px; overflow: hidden;\\" aria-hidden=\\"true\\"></div></div></div></div>"`;

exports[`<CreateRoomModal /> before create room success should render as snapshot 1`] = `"<div class=\\"ant-modal-root\\"><div class=\\"ant-modal-mask fade-appear\\"></div><div tabindex=\\"-1\\" class=\\"ant-modal-wrap \\" role=\\"dialog\\" aria-labelledby=\\"rcDialogTitle1\\"><div role=\\"document\\" style=\\"width: 520px;\\" class=\\"ant-modal zoom-appear\\"><div tabindex=\\"0\\" style=\\"width: 0px; height: 0px; overflow: hidden;\\" aria-hidden=\\"true\\"></div><div class=\\"ant-modal-content\\"><button type=\\"button\\" aria-label=\\"Close\\" class=\\"ant-modal-close\\"><span class=\\"ant-modal-close-x\\"><i aria-label=\\"icon: close\\" class=\\"anticon anticon-close ant-modal-close-icon\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"close\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z\\"></path></svg></i></span></button><div class=\\"ant-modal-header\\"><div class=\\"ant-modal-title\\" id=\\"rcDialogTitle1\\">Create a room</div></div><div class=\\"ant-modal-body\\"><span class=\\"ant-input-affix-wrapper\\"><span class=\\"ant-input-prefix\\"><i aria-label=\\"icon: user\\" class=\\"anticon anticon-user\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"user\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z\\"></path></svg></i></span><input placeholder=\\"Please enter Candidate name\\" type=\\"text\\" class=\\"ant-input\\" value=\\"\\"></span></div><div class=\\"ant-modal-footer\\"><div><button type=\\"button\\" class=\\"ant-btn\\"><span>Cancel</span></button><button disabled=\\"\\" type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Create</span></button></div></div></div><div tabindex=\\"0\\" style=\\"width: 0px; height: 0px; overflow: hidden;\\" aria-hidden=\\"true\\"></div></div></div></div>"`;
